### PR TITLE
Add utf8 memo program

### PR DESCRIPTION
**Pull request by CriesofCarrots**
New, simpler version, without Instruction.
Resulting program file is about 2kb smaller.

### CODE REVIEW

File: memo/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+version = "0.1.0"
+description = "Solana Program Library Memo"
+authors = ["Solana Maintainers <maintainers@solana.com>"]
+repository = "https://github.com/solana-labs/solana-program-library"
+license = "Apache-2.0"
+edition = "2018"
+
+[dependencies]
+num-derive = "0.2"
+num-traits = "0.2"
```
Comment (garious) [inline memo/Cargo.toml:15]: Can you review these dependencies? Seems like `num-derive`, `num-traits`, and `thiserror` might no longer be needed.

File: memo/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+solana-sdk = { version = "=1.2.0", default-features = false, features=["program"] }
```
Comment (garious) [inline memo/Cargo.toml:16]: Why the hard `=`?

File: memo/program-id.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Memo1UhkJRfHyvLMcVucJwxXeuD728EqVDDwQDxFMNo
```
Comment (mvines) [inline memo/program-id.md:1]: for the convenience of Rust clients, maybe the program-id should live in a .rs file too?  I guess this is a more general SPL setup comment that we can deal with later

File: memo/Cargo.toml (hunk 2 above)
Comment (CriesofCarrots) [reply memo/Cargo.toml:14]: The precedent here (Token) is to pin to a particular version of the solana-sdk. We still need to work out release/deployment processes for the programs in this repo. But I do think the `=` is helpful from a readability perspective, assuming what appears here is meant to represent what is published to the chain.

Comment (garious) [reply memo/Cargo.toml:14]: We'll need to add release tags any time we're publishing to the chain.  Definitely can't assume what's in the master branch is meaningful. Can't really assume what's outside the Cargo.lock is too meaningful either.  `=` deps are bad for downstream consumers.

File: memo/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+version = "0.1.0"
+description = "Solana Program Library Memo"
+authors = ["Solana Maintainers <maintainers@solana.com>"]
+repository = "https://github.com/solana-labs/solana-program-library"
+license = "Apache-2.0"
+edition = "2018"
+
+[dependencies]
+solana-sdk = { version = "=1.2.0", default-features = false, features=["program"] }
+solana-sdk-bpf-test = { path = "../bin/bpf-sdk/rust/test", default-features = false }
```
Comment (garious) [inline memo/Cargo.toml:15]: Just looking at this name here, with the `test` suffix. Can this be put in `[dev-dependencies]`?

File: memo/src/processor.rs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+};
+use std::str::from_utf8;
+
+entrypoint!(process_instruction);
+fn process_instruction<'a>(
+    _program_id: &Pubkey,
+    _accounts: &'a [AccountInfo<'a>],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    from_utf8(instruction_data)
```
Comment (garious) [inline memo/src/processor.rs:13]: Seems more readable to me:

```rust
from_utf8(instruction_data).map_err(|_| ProgramError::InvalidInstructionData)?;
Ok(())
```

Maybe @mvines didn't notice the return value?

File: memo/Cargo.toml (hunk 4 above)
Comment (CriesofCarrots) [reply memo/Cargo.toml:15]: Ah, actually memo doesn't need that one at all either

File: memo/program-id.md (hunk 3 above)
Comment (CriesofCarrots) [reply memo/program-id.md:1]: Yeah, good idea. I'll leave as-is for now, but will move around when we rethink how we want to publish rust assets.

File: memo/src/processor.rs (hunk 5 above)
Comment (CriesofCarrots) [reply memo/src/processor.rs]: I prefer that as well. I'll try to sneak it by him again ;)

File: memo/src/processor.rs [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    use solana_sdk::{program_error::ProgramError, pubkey::Pubkey};
+
+    #[test]
+    fn test_utf8_memo() {
+        let program_id = Pubkey::new(&[0; 32]);
+
+        let string = b"letters and such";
+        assert_eq!(
+            Ok(()),
+            process_instruction(&program_id, &mut vec![], string)
```
Comment (garious) [inline memo/src/processor.rs:29]: `cargo clippy --tests`

**Comment by garious** [comment]
:shipit:

File: memo/src/processor.rs (hunk 5 above)
Comment (mvines) [reply memo/src/processor.rs]: 🙈 ok do it now!

**Pull request merged**